### PR TITLE
Fix documentation warnings and errors

### PR DESCRIPTION
**Pull request by jesper-friis**
# Description
Fix most warnings and errors from doxygen and sphinx. 

Improved generated documentation.

## Type of change
- [x] Bug fix & code cleanup
- [ ] New feature
- [x] Documentation update
- [ ] Test update

## Checklist for the reviewer
This checklist should be used as a help for the reviewer.

- [ ] Is the change limited to one issue?
- [ ] Does this PR close the issue?
- [ ] Is the code easy to read and understand?
- [ ] Do all new feature have an accompanying new test?
- [ ] Has the documentation been updated as necessary?

### CODE REVIEW

File: bindings/python/dlite-collection-python.i [hunk 1]
```diff
@@ -56,19 +56,24 @@ class Collection(Instance):
...
     `d`, specifying the datatype of the object.  The datatype may have the
     following values:
 
-      - None: object is an IRI.
-      - Starts with '@': object is a language-tagged plain literal.
-        The language identifier follows the '@'-sign.
-      - Otherwise: object is a literal with datatype `d`.
+    - None: object is an IRI.
+    - Starts with '@': object is a language-tagged plain literal.
+      The language identifier follows the '@'-sign.
```
Comment (francescalb) [inline bindings/python/dlite-collection-python.i:70]:
```suggestion
      The language identifier follows the '@'-sign. e.g. `text@en`
```

File: bindings/python/dlite-entity.i [hunk 2]
```diff
@@ -93,10 +93,23 @@ char** dlite_swig_istore_get_uuids()
 %}
 
 
-
 /* ---------
  * Dimension
  * --------- */
+%feature("docstring", "\
+A dimension represented by its name and description.
+Metadata can define a set of common dimension for its properties that
```
Comment (francescalb) [inline bindings/python/dlite-entity.i:101]:
```suggestion
Metadata can define a set of common dimensions for its properties that
```

File: bindings/python/dlite-entity.i [hunk 3]
```diff
@@ -93,10 +93,23 @@ char** dlite_swig_istore_get_uuids()
...
+are referred to by the property shapes.
```
Comment (francescalb) [inline bindings/python/dlite-entity.i:102]:
```suggestion
are referred to by the shape of each property.
```

File: bindings/python/dlite-entity.i [hunk 4]
```diff
@@ -93,10 +93,23 @@ char** dlite_swig_istore_get_uuids()
...
+
+Arguments:
+    name: Name that of dimension.
```
Comment (francescalb) [inline bindings/python/dlite-entity.i:105]:
```suggestion
    name: Name of dimension.
```

File: bindings/python/dlite-entity.i [hunk 5]
```diff
@@ -93,10 +93,23 @@ char** dlite_swig_istore_get_uuids()
...
+    description: Description of the dimension.
```
Comment (francescalb) [inline bindings/python/dlite-entity.i:106]:
```suggestion
    name: Dimension name.
    description: Dimension description.
```
to unify with attributes.

File: bindings/python/dlite-entity.i [hunk 6]
```diff
@@ -123,7 +136,31 @@ struct _DLiteDimension {
  * Property
  * -------- */
 %feature("docstring", "\
-Creates a new property with the provided attributes.
+Represents a property.
+All metadata must have one or more properties that defines the instances
```
Comment (francescalb) [inline bindings/python/dlite-entity.i:140]:
```suggestion
All metadata must have one or more properties that define the instances
```

File: bindings/python/dlite-entity.i [hunk 7]
```diff
@@ -123,7 +136,31 @@ struct _DLiteDimension {
...
+of the metadata.
+
+Arguments:
+    name: Name of the property.
```
Comment (francescalb) [inline bindings/python/dlite-entity.i:144]:
```suggestion
    name: Property name.
```

File: bindings/python/dlite-entity.i [hunk 8]
```diff
@@ -123,7 +136,31 @@ struct _DLiteDimension {
...
+    type: Property type. Ex: 'int', 'blob14', 'float64', 'ref'...
+    ref: Optional. URL to metadata. Only needed for `type='ref'`.
+    shape: Optional. Specifies the dimensionality of property.  If `shape`
+        is not given, the property is a scalar (dimensionality zero).
+        It should be an sequence of dimension names.
```
Comment (francescalb) [inline bindings/python/dlite-entity.i:149]:
```suggestion
        It should be a sequence of dimension names.
```

File: doc/getting_started/tutorial.md [hunk 9]
```diff
@@ -11,14 +15,18 @@ This tutorial shows the steps for creating and instantiating a Dlite entity, and
...
-## Writing a DLite Entity
-Let us write an Entity representing our solar panel measurement. We start by creating a `json` file called `myEntity.json`.
+Writing a DLite Entity
+----------------------
+Let us write an Entity representing our solar panel measurement.
+We start by creating a `json` file called `myEntity.json`.
 
 ### **Step 1**: Giving it a unique identifier
-Firstly, we must provide a unique identifier for our Entity. There are several ways of doing this:
+Firstly, we must provide a unique identifier for our Entity.
```
Comment (francescalb) [inline doc/getting_started/tutorial.md:24]:
```suggestion
First, we must provide a unique identifier for our Entity.
```

File: bindings/python/dlite-collection-python.i [hunk 10]
```diff
@@ -56,19 +56,24 @@ class Collection(Instance):
...
     `d`, specifying the datatype of the object.  The datatype may have the
     following values:
 
-      - None: object is an IRI.
-      - Starts with '@': object is a language-tagged plain literal.
-        The language identifier follows the '@'-sign.
-      - Otherwise: object is a literal with datatype `d`.
+    - None: object is an IRI.
+    - Starts with '@': object is a language-tagged plain literal.
+      The language identifier follows the '@'-sign. e.g. `text@en`
```
Comment (francescalb) [inline bindings/python/dlite-collection-python.i:70]: This is not clear. 
Do we write ´o='text', d='@en'´ or ´d='test@en´? As I suggested in my previous comment I think my example is not completely correct.

Comment (jesper-friis) [inline bindings/python/dlite-collection-python.i:70]:
```suggestion
      The language identifier follows the '@'-sign. e.g. `@en`
```

Comment (jesper-friis) [reply bindings/python/dlite-collection-python.i:70]: You are right. It should be ´o='text', d='@en'´. Corrected the example

**Review by francescalb — APPROVED** [review]

**Pull request merged**